### PR TITLE
Add Harvester node driver support

### PR DESCRIPTION
**Pull request by futuretea**
**Related issues**
https://github.com/harvester/harvester/issues/2140

### CODE REVIEW

**Comment by guangbochen** [comment]
link to https://github.com/rancher/terraform-provider-rancher2/issues/862

File: rancher2/schema_node_template.go [hunk 1]
```diff
@@ -12,15 +12,37 @@ type NodeTemplate struct {
...
+	"azure_config",
+	"digitalocean_config",
+	"harvester_config",
+	"hetzner_config",
+	"linode_config",
+	"opennebula_config",
+	"openstack_config",
+	"vsphere_config"}
+
+func getConflicts(fieldNames []string, fieldName string) []string {
```
Comment (thedadams) [inline rancher2/schema_node_template.go:36]: This function should be moved to the first commit so as to not break things like `git bicect` and `git revert`.

**Review by thedadams — CHANGES_REQUESTED** [review]

File: rancher2/schema_node_template.go (hunk 1 above)
Comment (futuretea) [reply rancher2/schema_node_template.go:36]: @thedadams Updated

**Review by thedadams — APPROVED** [review]

**Comment by guangbochen** [comment]
Hi team, can we please have more reviews or this can be merged directly? thanks in advance.

**Review by jakefhyde — APPROVED** [review]

**Comment by snasovich** [comment]
@guangbochen @futuretea , have 2 approvals so going to merge it as I'm not sure you have permissions. Please let me know if for any reason it shouldn't have been merged.

**Comment by guangbochen** [comment]
@snasovich we don't have the merge permission, can u please help to merge it, thanks.

**Comment by snasovich** [comment]
@guangbochen , done.

**Pull request merged**